src/Fmt_ast.ml

```diff
@@ -166,44 +156,51 @@ let maybe_disabled_k c (loc : Location.t) l f k =
             drop_while s ~f:(fun i c ->
                 Char.is_whitespace c && i < indent_of_first_line ) )
     in
-    k (list l "@\n" (fun s -> str s))
+    k (list l "@\n" str)
 
 let maybe_disabled c loc l f = maybe_disabled_k c loc l f Fn.id
 
 let update_config_maybe_disabled c loc l f =
-  let c = update_config c l in
-  maybe_disabled c loc l f
+  maybe_disabled (update_config c l) loc l f
```

Comment:
It is technically equivalent here, but the previous form is preferable I think since it emphasizes the stateful/monadic nature of update_config, and can be written the same everywhere in the previous form, where in the current one we need to remember to undo this change as soon as another use of `c` is added.

Reply:
Fixed.